codegen/core/src/main/scala/com/lightbend/akkasls/codegen/ModelBuilder.scala

```diff
@@ -109,7 +109,19 @@ object ModelBuilder {
       viewId: String,
       transformedUpdates: Iterable[Command]
   ) extends Service(fqn, commands) {
-    require(fqn.name.endsWith("View") || fqn.name.endsWith("Impl"))
+
+    val viewClassName =
+      if (fqn.name.endsWith("View")) fqn.name
+      else fqn.name + "View"
+
+    val abstractViewName = "Abstract" + viewClassName
+    val handlerName = viewClassName + "Handler"
+    val providerName = viewClassName + "Provider"
+
+    val classNameQualified = s"${fqn.parent.javaPackage}.$viewClassName"
+    val providerNameQualified = s"${fqn.parent.javaPackage}.$providerName"
+
+    val state = State(commands.head.outputType)
```

Comment:
I think just picking the first command output type as the state type is not quite safe, one scenario where it will break that I can think of is a view with no transform and only a query that "projects" the result into some other type (unless I remember wrong that we support that).

Reply:
we have a validation for the return types, if that is the case you think about: AS-00109 All View update methods must have the same return type

Reply:
Hmm, maybe I'm misinterpreting that the `commands` here will include both queries and update method - was thinking that the first command (not sure how they are ordered) could be the query, and that could be a projection into some non-state type.

Reply:
ah, I understand, you are right if commands includes both update methods and query methods. it should look for a update method

Reply:
Did something about that now, looking only at the first update method (with some test coverage). Not sure if there may still be cases I have not thought about.